### DISCUSSION TITLE
Kratos release policy is a disaster

### DISCUSSION THREAD

**Discussion by nerrixde** [ory/kratos#2886] [Q&A] [answered]
I can absolutely understand that developing new features takes time.

Extensive testing - especially in such sensitive areas - is important and right.

I'm not asking for any features either. I also know that you have other things to do.

But what I really don't find acceptable is that completely finished and marked features are unreleased for almost half a year now. 

There were numerous problems that arose in the wonderful release wave with 2fa and co. that are even fixed, but are not part of v0.10.1. 
I don't know what's the point of delaying finished features like that, and it's starting to interfere with production. 

So Kratos is abandoned? Is it expected to pick the appropriate commit from Master and deploy it yourself?

Or should we all just switch to Keykloak? I don't know what's the grand scheme here, I didn't get an answer in Slack either.

**Comment by aeneasr** [comment, marked as answer]
Hi @nerrixDE , sorry to see you so frustrated. We are planning a release of Ory Kratos before years end and are working on a policy for releasing open source as we have been slow in doing so this year as our focus was on other things (https://github.com/ory/hydra/releases/tag/v2.0.0, https://github.com/ory/keto/releases/tag/v0.10.0-alpha.0, ...).

You may always, at all times, build the binaries yourself with:

```
make docker
# or
go install -tags sqlite,json1 .
```

Whether or not you decide to switch to Keycloak is up to you. The Apache 2.0 license sums it up:

> Unless required by applicable law or agreed to in writing, licensor provides the work (and each contributor provides its contributions) on an "as is" basis, without warranties or conditions of any kind, either express or implied, including, without limitation, any warranties or conditions of title, non-infringement, merchantability, or fitness for a particular purpose.

Using open source is the privilege of using work done by others without having to give anything back. Please keep that in mind when interacting with maintainers. You can help Ory a lot by using the commercial open source service ( https://console.ory.sh ). It helps keep the lights on, hire more people to work on Ory open source software, and so on. Thanks!

**Reply by nerrixde** [reply]
I absolutely understand your point. Thats why I mentioned my respect for your priorities. 
**However**: It's hard to grasp how it takes so much resources or time to release existing and merged (!) code. Especially bugfixes with webhooks are already live in master, but not released, for months now!
For sure, take your time, I absolutely agree with you that open source maintainers owe nothing, I share this view myself! 
Yet I think there is a difference when asking for new features or "stuff to be developed" vs. just releasing already done and merged work. 
But thank you for announcing the release before 2023, that is greatly appreciated. Kratos is a great project and in my view superior to other solutions out there. I am very confident there will be corperate FOSS contributions coming once it get's even more production-ready.

**Reply by aeneasr** [reply]
Thanks @nerrixDE ! We will probably release up to verification codes as a minor release without breaking changes. For codes we still want to get both recovery and verification codes in one Go to prevent upgrade problems :)
